### ISSUE TITLE
Reduce file size of built apps

### ISSUE THREAD

**Issue by ruarch** [eclipsesource/tabris-js#539]
Any reason why apps file size are around 10MB

**Comment by hstaudacher** [comment]
ios or android or both?

**Comment by ruarch** [comment]
android

**Comment by mpost** [comment]
The Tabris,js Android client bundles the V8 JavaScript engine to execute JavasScript code. Currently we bundle the V8 engine for each supported platform (arm, x86 in 32/64 bit) in the Android apk. This has the convenience that you can run the apk everywhere but of course it inflates the file size.

We use the V8 engine packaged by J2V8 and there is an ongoing discussion happening on how to make the V8 more easily consumable (https://github.com/eclipsesource/J2V8/issues/77) which will allow us to build (smaller) apks for individual architectures.

**Comment by ruarch** [comment]
ok.

**Comment by mpost** [comment]
@ralfstx I wouldn't consider the issue closed. It is a valid concern and we should keep it open until resolved.

**Comment by ralfstx** [comment]
@mpost Right, I agree.

**Comment by irbull** [comment]
I will look at how we can repackage J2V8 so we only ship the required natives with each platform. Once this is complete we will need to update the tabris-android builds to produce different APKs.

**Comment by calebeaires** [comment]
+1

**Comment by djMax** [comment]
+1

**Comment by irbull** [comment]
With the help of @mpost, I've packaged J2V8 as an _aar_ (Android Archive?). This means we don't need as many variants, or as many different native bits in the jar.  
```
<dependency>
  <groupId>com.eclipsesource.j2v8</groupId>
  <artifactId>j2v8</artifactId>
  <version>3.0.6</version>
  <type>aar</type>
</dependency>
```
This is available with 3.0.6 of J2V8 and I think @mpost is now consuming this in the tabris-android client.
